modules/subscriptions/subscriptions/subscriptions-settings.tsx

```diff
@@ -17,8 +17,49 @@ import Button from "@/resources/containers/button";
 
 const messageHandler = MessageHandler.get();
 
+const SubscriptionValidity: React.FC<{ subscriptionId: string }> = ({ subscriptionId }) => {
+  const [validityDate, setValidityDate] = useState<string | null>(null);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    const fetchSubscriptionData = async () => {
+      try {
+        const data = await Service.require("subscriptions").getSubscriptionById(subscriptionId);
+        if (!data?.current_period_end) {
+          setValidityDate(t("subscription-settings.invalid-date"));
+          return;
+        }
+        const date = typeof data.current_period_end === 'number'
+          ? moment.unix(data.current_period_end)
+          : moment(data.current_period_end);
+        if (date.isValid()) {
+          setValidityDate(date.format("DD MMM, YYYY"));
+        } else {
+          console.error("Invalid date received:", data.current_period_end);
+          setValidityDate(t("subscription-settings.invalid-date"));
+        }
+      } catch (error) {
+        console.error("Error fetching subscription data:", error);
+        setValidityDate(t("subscription-settings.error-fetching"));
+      }
+    };
+    fetchSubscriptionData();
```

Comment:
The Date  solution may work 

```
  const date = typeof data.current_period_end === 'number'
          ? moment.unix(data.current_period_end)
          : moment(data.current_period_end);
          ```

But we are calling the API for each single record, when we actually got load this data from the API. 
We shoudt need to call the API, we can just use the exiting object , can not push this into main.

Get directly the object loaded and do the check

Reply:
I agree @abelbordonado if we could have it in one API it would be much better.